File: Easy/367/367.py
class Solution:
    def isPerfectSquare(self, num: int) -> bool:
        i, j = 0, num + 1
        while i < j:
            mid = i + (j - i) // 2
            p = mid * mid
            if p == num:
                return True
            elif p < num:
                i = mid + 1
            else:
                j = mid
        return False

File: Easy/367/367.java
class Solution {
    public boolean isPerfectSquare(int num) {
        int i = 0;
        int j = num;
        while (i <= j) {
            int mid = i + (j - i) / 2;
            if ((long) mid * mid == num) {
                return true;
            } else if ((long) mid * mid < num) {
                i = mid + 1;
            } else {
                j = mid - 1;
            }
        }
        return false;
    }
}